checkers/Board.py

```diff
@@ -83,77 +84,34 @@ def find_valid_pieces(self, player_turn):
-                left_grid_row_, left_grid_col_ = left_grid_row + player_turn, left_grid_col - 1
-                if left_grid_row_ >= 0 and left_grid_col_ >= 0 and left_grid_col_ <= 7 and (
-                        self.grid[left_grid_row][left_grid_col].player !=
-                        player_turn) and (
-                            self.grid[left_grid_row_][left_grid_col_] == 0):
-                    self.grid[curr_row][curr_col].valid_grids.append(
-                        (left_grid_row_, left_grid_col_))
-                    self.opponent_piece_eaten['left'] = [
-                        left_grid_row, left_grid_col, left_grid_row_,
-                        left_grid_col_
-                    ]
-
-            else:
-                self.grid[curr_row][curr_col].valid_grids.append(
-                    (left_grid_row, left_grid_col))
-
-        # check to make sure between boundaries
-        if right_grid_row >= 0 and right_grid_row <= 7 and right_grid_row <= 7 and right_grid_col <= 7:
-
-            if isinstance(self.grid[right_grid_row][right_grid_col], Piece):
-                right_grid_row_, right_grid_col_ = right_grid_row + player_turn, right_grid_col + 1
-                if right_grid_row_ <= 7 and right_grid_col_ >= 0 and right_grid_col_ <= 7 and (
-                        self.grid[right_grid_row][right_grid_col].player !=
-                        player_turn) and (
-                            self.grid[right_grid_row_][right_grid_col_] == 0):
-                    self.grid[curr_row][curr_col].valid_grids.append(
-                        (right_grid_row_, right_grid_col_))
-                    self.opponent_piece_eaten['right'] = [
-                        right_grid_row, right_grid_col, right_grid_row_,
-                        right_grid_col_
-                    ]
-
-            else:
-                self.grid[curr_row][curr_col].valid_grids.append(
-                    (right_grid_row, right_grid_col))
-
-        if 'left' in self.opponent_piece_eaten and 'right' not in self.opponent_piece_eaten:
-            print("Left side exist and right not")
-            self.grid[curr_row][curr_col].valid_grids = [
-                self.grid[curr_row][curr_col].valid_grids[0]
-            ]
-
-        if 'right' in self.opponent_piece_eaten and 'left' not in self.opponent_piece_eaten:
-            print("Right side exist and left not")
-            self.grid[curr_row][curr_col].valid_grids = [
-                self.grid[curr_row][curr_col].valid_grids[-1]
-            ]
+        curr_row, curr_col = (
+            row,
+            col) if row is not None and col is not None else self.selected_piece
+
+        # set is used here to avoid addition of same piece multiple times
+        self.piece_set = {(curr_row, curr_col)}
+        self.piece_free_grids = []
+
+        # dict - {(after_captured_grid): [(capture_piece), (parent_grid)]}
+        self.capture_pieces = {}
```

Comment:
Shouldn't these instance attributes be declared inside the constructor first? @mlsdpk

Reply:
Yes, sure. I will update it accordingly.